### PR TITLE
Include valuators group in Investment table & csv download

### PR DESCRIPTION
**Pull request by raul-fuentes**
Where
=====
* **Related Issue:** https://github.com/consul/consul/issues/2472

What
====
- In the Budgets Investments admin section, add's the valuators group in the html table and in the csv

How
===
- Added one more column to the self.to_csv function on Budget::Investment and app/views/admin/budget_investments/_investments.html.erb

Screenshots
===========
- Html
screenshot from 2018-02-21 13-38-32
- Csv
screenshot from 2018-02-21 13-41-06

Test
====
- Added more conditions to the existing tests

Deployment
==========
- none

Warnings
========
- none

### CODE REVIEW

**Comment by bertocq** [comment]
@raul-fuentes some comments about this PR:
- A valuator can be assigned to an Investment either directly or through a group. You're not testing the scenario where a valuator is assigned to a group, and the group to an investment (without the valuator being  assigned individually to the group) and that may be covering some issues in the code. Like line https://github.com/AyuntamientoMadrid/consul/pull/1287/files#diff-e62910e895a38477491f27ec01e9c61dR79 or https://github.com/AyuntamientoMadrid/consul/pull/1287/files#diff-8e806f1abf6a06e54748ebcb2ce63d3aR372 ... the fact that there are no individual valuator assignments doesn't have anything to do with the fact that there might be group valuator assignments 

- You're using valuator's valuator_groups instead of directly investment.valuator_groups... so I think I can confirm there might be a missunderstanding of how the feature works.


- There is no need to add translations to any other language but English and Spanish. Translations will be managed through crowdin project in the future. Also if there is no translation its better than adding a default one in english.

- Please 🙏  remember to user rubocop either as a plugin of your favourite IDE/text editor, or as a command line util. I've a not-yet-polished section for Rubocop usage here that may clarify things https://github.com/consul/consul/wiki/Rubocop

- I see two times `investment.valuators.map{|val| val.valuator_group.try(:name)}.join(', ')` (first in a function that encloses it 👌 but the second one could just be a call to that function)

- `.size == 0` can also be just `.zero?` but I think its just a readability thing, not a performance improvement.

**Comment by raul-fuentes** [comment]
* Asking Maria, I understood that the groups of the assigned evaluators were shown and not all the groups, both those assigned to the investment and the evaluators.

* As for the function and the zeros, I immediately change them.

* Should I remove translations other than Spanish and English?

**Comment by bertocq** [comment]
- I'm sorry talking to maria mislead you. The clear requirement would be "When downloading a list of investments, we should include a new column that for each investments contains all the assigned ValuatorGroups, in the same fashion we show the assigned Valuators". If in doubt about how ValuatorGroup assignments work just check the Investment & Admin::BudgetInvestmentsController code, but to keep things short: Valuator assignments & ValuatorGroup assignments are totally independent... we could have "John" assigned to an investment both directly and through one or more ValuatorGroups he belongs to

- Yes please only en & es, no need for hebrew or even valencian, translations revisions go in another service & group of reviewers.

File: app/models/budget/investment.rb [hunk 1]
```diff
@@ -363,20 +364,17 @@ def self.to_csv(investments, options = {})
                   else
                     I18n.t("admin.budget_investments.index.no_admin_assigned")
                   end
-          vals = if investment.valuators.empty?
-                   I18n.t("admin.budget_investments.index.no_valuators_assigned")
-                 else
-                   investment.valuators.collect(&:description_or_name).join(', ')
-                 end
+          vals = ApplicationController.helpers.assigned_valuators(investment)
```
Comment (bertocq) [inline app/models/budget/investment.rb]: Calling a ApplicationController from a model doesn't feel quite right for me.

Comment (bertocq) [reply app/models/budget/investment.rb]: Maybe the whole `to_csv` method should be extracted to a class that encloses the logic... something like BudgetInvestmentCSVExporter... I don't know ... like a ServiceObject?

Comment (raul-fuentes) [reply app/models/budget/investment.rb]: This is the only way I have found to share the function that returns the names, between the model and the view. in any other case I would have to repeat code.

Maybe I could move the calculation of the contents of the table to a service object and call it both in the view and in the csv.

I'm going to see if it works better

Comment (bertocq) [reply app/models/budget/investment.rb]: Well if the problem is just `assigned_valuation_groups` and `assigned_valuators` there's a really easy fix for it: Move those two functions on to the Budget::Investment model. The `to_csv

Comment (raul-fuentes) [reply app/models/budget/investment.rb]: :sob: I'm dumb...

I'll fix in a moment...

**Comment by bertocq** [comment]
Raul I added this changes https://github.com/AyuntamientoMadrid/consul/pull/1287/commits/6584c3ffb54e12667e0c2aad55187bb6cf96b51c there is no need for Model layer to know View stuff (I18n strings) if possible. I know to_csv has some and that burns me from the inside, and we'll fix it sooner or later, but existing bad practices doesn't allow for more.

Also, english translations should not be literal, always help yourself with existing ones if you don't have a clear idea of how to proceed or just ask for help.

**Comment by bertocq** [comment]
Only flakies, lets go

**Comment by raul-fuentes** [comment]
As I said [here](https://github.com/AyuntamientoMadrid/consul/pull/1265#issuecomment-369759017), I was not aware of the side effects of I18n out of the views, thanks for the advise

**Comment by bertocq** [comment]
Its not a side effect or inconvenience. MVC is the separation of what a model should know and what the View should care about. Mixing both is never a good idea, but a clear smell that something is mixed up

**Pull request merged**